src/v2i-hub/ODEForwardPlugin/src/ODEForwardPlugin.cpp

```diff
@@ -147,14 +147,16 @@
         std::sprintf(teststring, "{\"BsmMessageContent\":[{\"metadata\":{\"utctimestamp\":\"%s\"},\"payload\":\"%s\"}]}",getISOCurrentTimestamp<chrono::microseconds>().c_str(),routeableMsg.get_payload_str().c_str());
 
         //  check for schedule
-        if(_freqCounter++%_scheduleFrequency == 0)
-                QueueKafkaMessage(kafka_producer, _BSMkafkaTopic, teststring);	
-	
+        if(_freqCounter++%_scheduleFrequency == 0) {
+			QueueKafkaMessage(kafka_producer, _BSMkafkaTopic, teststring);
+		}
+		
+		delete teststring;
```

Comment:
@jwillmartin Sorry to keep finding new things on this!  Sonar says "Memory allocated by 'new[]' should be deallocated by 'delete[]', not 'delete'" and claims segfaults can happen if the wrong delete is used.  You can check out their comments at https://sonarcloud.io/project/issues?resolved=false&sinceLeakPeriod=true&types=BUG&pullRequest=459&id=usdot-fhwa-ops_V2X-Hub&open=AYVZ04rg0fVLKmkUMLZQ

Reply:
Ha, all good. Fixed!